src/background/services/accounts/AccountsService.ts

```diff
@@ -98,8 +100,27 @@ export class AccountsService implements OnLock, OnUnlock {
     // refresh addresses so in case the user switches to testnet mode,
     // as the BTC address needs to be updated
     this.networkService.developerModeChanged.add(this.onDeveloperModeChanged);
+
+    // TODO(@meeh0w):
+    // Remove this listener after E-upgrade activation on Fuji. It will be no longer needed.
+    this.networkService.uiActiveNetworkChanged.add(
+      this.#onActiveNetworkChanged
+    );
   }
 
+  #wasDevnet = false;
+
+  #onActiveNetworkChanged = async (network?: Network) => {
+    if (!network) {
+      return;
+    }
+
+    if (isDevnet(network) !== this.#wasDevnet) {
```

Comment:
_Note to self: can this become a race-condition problem between `developerModeChanged` and `uiActiveNetworkChanged`?_